Gems/Atom/RPI/Code/Source/RPI.Public/VisibleObjectContext.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <Atom/RPI.Public/VisibleObjectContext.h>
+#include <Atom/RPI.Reflect/Base.h>
+#include <AzCore/Debug/Profiler.h>
+#include <AzCore/std/sort.h>
+
+AZ_DECLARE_BUDGET(RPI);
+
+namespace AZ
+{
+    namespace RPI
+    {
+        void VisibleObjectContext::Shutdown()
+        {
+            m_visibleObjectListContext.Clear();
+
+            m_finalizedVisibleObjectList.clear();
+        }
+
+        void VisibleObjectContext::AddVisibleObject(const void* userData, float depth)
+        {
+            if (Validation::IsEnabled())
+            {
+                if (!userData)
+                {
+                    AZ_Error("VisibleObjectContext", false, "Null userData was added to a VisibleObjectContext. This is not permitted and will crash if validation is disabled.");
```

Comment:
That's a good question. I copied this pattern from the DrawListContext. I suspect that whoever wrote it initially was maybe just trying to be conservative and avoid an extra null check when validation is disabled since this is the inner-loop of culling. This probably falls under the bucket of premature optimization, unless they actually noticed a difference in performance.

Reply:
I am approving the PR but if you think there is a chance it will crash we should return even at the cost of adding an if check.